### PR TITLE
initial notebooks ga updates

### PR DESCRIPTION
**Pull request by benvenker**
## Test plan

View docs on docs.sourecegraph.com to verify they are correct.

### CODE REVIEW

**Comment by novoselrok** [comment]
We probably also want to remove `(beta)` from the sidebar here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/doc/sidebar.md?L37:4

File: doc/notebooks/index.md [hunk 1]
```diff
@@ -32,30 +32,68 @@ body.theme-dark .markdown-body ul li:before {
...
+
+### Web-based notebooks
+The simplest way to get started with Notebooks is to create one using the web interface. Notebooks created this way have the advantage of being interactive, letting you see the content of your blocks in realtime as you create your notebook. You can also utilize the [notepad](../notebooks/notepad.md) to dramatically speed up notebook creation.
+
+You can also create web-based notebooks by importing plain old Markdown files and then augmenting them with Sourcegraph notebook block types in the web interface. A new notebook will automatically be created when you import a standard markdown file. From there, you can modify it however you like in the web interface.
+
+Web-based notebooks are automatically saved as they're edited. There is currently no version control, version history, or versioning system.
+
+### File-based notebooks
+Alternatively, you can create notebooks using text files with the `.md.snb` file extension. These files are rendered specially by Sourcegraph (either on sourcegraph.com or within your Sourcegraph instance) to display notebook blocks alongside standard Markdown blocks.
```
Comment (novoselrok) [inline doc/notebooks/index.md:54]: The extension is `.snb.md`, not `.md.snb.`. There are also a couple of instances below that will have to be changed.

File: doc/notebooks/index.md [hunk 2]
```diff
@@ -32,30 +32,68 @@ body.theme-dark .markdown-body ul li:before {
...
+
+Whenever you view a `.md.snb` file on Sourcegraph, you'll see a "Run all blocks" button near the top of the notebook, which will execute all the notebook blocks at once. Markdown and file blocks are rendered by default.
+
+File-based notebooks have the advantage of living anywhere you store text files. The disadvantage comes during composition, as you won't be able to see the contents of your blocks while you create your notebook.
+
+### Combined approaches
+#### Compose online and export to disk
+If you prefer to keep your notebooks in your repos but want to compose them on the web, you can get the best of both worlds by composing your notebooks on your sourcegraph instance and then exporting them to your
```
Comment (novoselrok) [inline doc/notebooks/index.md:62]: Missing the ending?

File: doc/notebooks/index.md [hunk 3]
```diff
@@ -32,30 +32,68 @@ body.theme-dark .markdown-body ul li:before {
...
+Whenever you view a `.md.snb` file on Sourcegraph, you'll see a "Run all blocks" button near the top of the notebook, which will execute all the notebook blocks at once. Markdown and file blocks are rendered by default.
+
+File-based notebooks have the advantage of living anywhere you store text files. The disadvantage comes during composition, as you won't be able to see the contents of your blocks while you create your notebook.
+
+### Combined approaches
+#### Compose online and export to disk
+If you prefer to keep your notebooks in your repos but want to compose them on the web, you can get the best of both worlds by composing your notebooks on your sourcegraph instance and then exporting them to your 
+
+#### Embed notebooks anywhere
+Sourcegraph notebooks can be [embedded](../notebooks/notebook-embedding.md) anywhere that allows iFrames. Notebooks hosted on sourcegraph.com can be embedded anywhere. Notebooks hosted on your private instance are subject to your organization's security policies, but can generally be viewed by any user with access to your instance as long as they're logged in.
```
Comment (novoselrok) [inline doc/notebooks/index.md:65]: The F in `iFrames` doesn't have to be capitalized.

File: doc/notebooks/index.md [hunk 4]
```diff
@@ -32,30 +32,68 @@ body.theme-dark .markdown-body ul li:before {
...
+
+## Notebook blocks
+The currently supported block types are:
+- Query
```
Comment (novoselrok) [inline doc/notebooks/index.md:69]: This list renders a bit weird:

Screenshot 2022-04-19 at 13 41 42

You might have to add an empty new line before the first list item.

File: doc/notebooks/index.md [hunk 5]
```diff
@@ -32,30 +32,68 @@ body.theme-dark .markdown-body ul li:before {
...
+- File
+- Symbol
+- Markdown
+
+[Read more about block types](../notebooks/blocks.md).
+
+## Searching notebooks
+Notebooks created through the web interface are full text searchable from the /notebooks page. Each tab has its own search box and each search box is scoped to that tab. The exception is the Explore tab, which searches all notebooks you have access to.
+
+Searches will match on notebook titles and any text in Markdown blocks. Searching through other block types is not supported because they are dynamic in nature.
```
Comment (novoselrok) [inline doc/notebooks/index.md:79]: Not just text in Markdown blocks, but any text in any of the blocks.

File: doc/notebooks/blocks.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+## Markdown blocks
+Markdown blocks support standard Markdown formatting, enabling you to create headings, lists, code blocks, and more. They are the foundational block type for providing additional context to the specialized block types described below.
+
+## Query blocks
+Query blocks support the full Sourcegraph search query language, allowing you to use any of our search types, filters, patterns, and predicates to display the most relevant results.
+
+## Symbol blocks
+With symbol blocks, you can identify the symbol you want to highlight. Symbol blocks are special. As long as the symbol definition stays within the file you selected when you created the block, you don't need to update it when the file changes. Symbol blocks "follow" the symbol around the file and so always display its current location.
+
+Symbol blocks are a subset of the query block, providing some UI affordances to make it easier to find symbols, such as a default `type:symbol` in the query, as well as a purpose-built typeahead specialized for symbol selection.
```
Comment (novoselrok) [inline doc/notebooks/blocks.md:14]: I wouldn't say that symbol blocks are necessarily a subset of the query block. I would just mention that they can use the Sourcegraph query language to identify the symbol they want to track.

File: doc/notebooks/blocks.md [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+## File blocks
+File blocks are similar to symbol blocks in that they are some special affordances to make them easier to create. You can add an entire file the file block, or you can select a line range of a file. File ranges are great for embedding code snippets into a notebook or highlighting important files. File blocks are editable so you can modify a full file to only show a line range from it, or remove the line range to show an entire file.
+
+If you're viewing a file in Sourcegraph search, you can also copy the URL and paste it directly into a file block. If you have a line range selected it will be preserved.
```
Comment (novoselrok) [inline doc/notebooks/blocks.md:19]: They can also paste the URL into the command palette.

**Review by novoselrok — APPROVED** [review]

File: doc/notebooks/index.md (hunk 2 above)
Comment (benvenker) [reply doc/notebooks/index.md:62]: Ha, yes that's where I left off yesterday afternoon I think. Wanted to push up before I got on a plane :D

File: doc/notebooks/index.md (hunk 4 above)
Comment (benvenker) [reply doc/notebooks/index.md:69]: Weird...there was one but added another. We'll see...

File: doc/notebooks/blocks.md [hunk 8]
```diff
@@ -0,0 +1,19 @@
+Blocks are the compositional units of a notebook. You can interleave the various block types in a notebook to create rich, powerful documentation. There are four supported block types.
+
+# Block types
+
+## Markdown blocks
+Markdown blocks support standard Markdown formatting, enabling you to create headings, lists, code blocks, and more. They are the foundational block type for providing additional context to the specialized block types described below.
+
+## Query blocks
+Query blocks support the full Sourcegraph search query language, allowing you to use any of our search types, filters, patterns, and predicates to display the most relevant results.
```
Comment (lguychard) [inline doc/notebooks/blocks.md:9]:
```suggestion
Query blocks support the full Sourcegraph search query language, allowing you to use our full [search syntax](../code_search/reference/index.md), including any of our types, filters, patterns, and predicates, to display the most relevant results.
```

Suggestion, non-blocking: link out to our search syntax docs. Link in the suggestion should be correct 🙂

File: doc/notebooks/index.md (hunk 5 above)
Comment (benvenker) [reply doc/notebooks/index.md:79]: Updated to reflect this clearly.

File: doc/notebooks/index.md [hunk 9]
```diff
@@ -32,30 +32,68 @@ body.theme-dark .markdown-body ul li:before {
...
 
-To try out notebooks on your enterprise install, enable them in global settings:
+## Notebook types
+Notebooks can be created in two ways. Through the web interface or via special Markdown files with the special `.md.snb` extension. To view file-based notebooks you must view the files in the file view on sourcegraph.com or on your Sourcegraph instance.
+
+### Web-based notebooks
+The simplest way to get started with Notebooks is to create one using the web interface. Notebooks created this way have the advantage of being interactive, letting you see the content of your blocks in realtime as you create your notebook. You can also utilize the [notepad](../notebooks/notepad.md) to dramatically speed up notebook creation.
+
+You can also create web-based notebooks by importing plain old Markdown files and then augmenting them with Sourcegraph notebook block types in the web interface. A new notebook will automatically be created when you import a standard markdown file. From there, you can modify it however you like in the web interface.
```
Comment (lguychard) [inline doc/notebooks/index.md:49]:
```suggestion
You can also create web-based notebooks by importing plain Markdown files and then augmenting them with Sourcegraph notebook block types in the web interface. A new notebook will automatically be created when you import a standard markdown file. From there, you can modify it however you like in the web interface.
```

Nit: colloquialism

**Review by lguychard — APPROVED** [review]

**Pull request merged**